### PR TITLE
Replaces DeveloperErrors with Checks in Renderer folder

### PR DESCRIPTION
**Pull request by ateshuseyin**

### CODE REVIEW

**Comment by pjcozzi** [comment]
Thanks @ateshuseyin, looks good at quick glance!

Does anyone want to review more closely and test?  @moneimne?

File: Source/Renderer/Buffer.js [hunk 1]
```diff
@@ -7,6 +7,7 @@ define([
         '../Core/DeveloperError',
         '../Core/IndexDatatype',
         '../Core/WebGLConstants',
+        '../Core/Check',
```
Comment (moneimne) [inline Source/Renderer/Buffer.js]: Make sure that you add the define paths in alphabetical order. Same applies to other files.

File: Source/Renderer/Buffer.js [hunk 2]
```diff
@@ -16,6 +17,7 @@ define([
         DeveloperError,
         IndexDatatype,
         WebGLConstants,
+        Check,
```
Comment (moneimne) [inline Source/Renderer/Buffer.js]: Make sure that you add the defines here in alphabetical order, too. Same applies to other files.

File: Source/Renderer/Buffer.js [hunk 3]
```diff
@@ -7,6 +7,7 @@ define([
         '../Core/DeveloperError',
```
Comment (moneimne) [inline Source/Renderer/Buffer.js]: Remove `DeveloperError` from defines. Same applies to other files.

**Comment by moneimne** [comment]
All the tests seem to pass! Just added some small comments for cleaning up the defines at the top of the files.

**Comment by pjcozzi** [comment]
Thanks for the review, @moneimne!

**Comment by ateshuseyin** [comment]
I fixed issues from the review and replace more. There is some problems with if clause checks. For example in Context.js:~896 there is if clause checks like below;
```
if (count < 0) {
            throw new DeveloperError('drawCommand.count must be greater than or equal to zero.');
} 
```
When I change it into;
```
Check.typeOf.number.greaterThanOrEquals('drawCommand.count', count, 0);
```

It causes to hundreds of failed spec. Because count is  `undefined` in some cases and `count < 0`is false but checking with `Check` causes to `DeveloperError`

**Comment by pjcozzi** [comment]
@moneimne can you please suggest a solution here?  Updates to `Check` are OK.

**Comment by moneimne** [comment]
As long as `Check` is okay, could you add another one before it to make sure `count` is defined first?

**Comment by ateshuseyin** [comment]
The problem is about old code, not the Check. 
The result of count < 0 is false for the undefined count. So old code guess undefined is greater than or equal to zero. Existing specs too. But not.  When I replace all appropriate DeveleporErrors by Check, causes about 1000 spec fail.

**Comment by moneimne** [comment]
Ah, I see what you mean now. It seems like the code only wants to throw an error if `count` is less than 0. If `count` is undefined, it is being set to a default value in the next few lines of code. 

It would be best to use `DeveloperError` instead of `Check` in these cases. So you could change the condition to `if (defined(count) && count < 0)` to make it more clear, then throw a `DeveloperError`.

**Comment by ateshuseyin** [comment]
deal

**Comment by ateshuseyin** [comment]
There is no error on my local but one spec fails on travis

**Comment by pjcozzi** [comment]
@ateshuseyin the test likely fails in CI since the WebGL instancing extension is support on your machine but **not** supported in CI.  I suspect the test needs to be updated.

> 1) throws when instanceCount is greater than one and the instanced arrays extension is not supported
     Renderer/Draw
     Expected function to throw DeveloperError.
    at Specs/Renderer/DrawSpec.js:1376:16
    at Object.<anonymous> (Specs/customizeJasmine.js:75:30)

**Comment by pjcozzi** [comment]
@ateshuseyin do you still plan to look at the test failure?  I provided some ideas above.

**Comment by ateshuseyin** [comment]
I will. I was in holiday :)

**Comment by hpinkos** [comment]
I cleaned up the PR and fixed the failing test.  We just had a bad test case.

@ggetz can you review?

**Pull request merged**